.github/workflows/build.yaml

```diff
@@ -14,7 +17,8 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: "1.20"
+          go-version: ${{ env.GO_VERSION }}
```

Comment:
This should be updated if https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/264 is merged before this PR.